src/Rest/RestRequest.cs

```diff
@@ -11,22 +11,22 @@ internal RestRequest(string resource, HttpMethod method)
             Resource = resource;
         }
 
-        public Dictionary<string, string> QueryParameters { get; private set; } = new Dictionary<string, string>();
-        public Dictionary<string, string> Headers { get; private set; } = new Dictionary<string, string>();
-        public HttpMethod Method { get; private set; }
-        public string Resource { get; private set; }
-        public string JsonBody { get; private set; }
-        public System.IO.Stream FileStream { get; private set; }
-        public string FileStreamContentType { get; private set; }
-        public string FileStreamName { get; private set; }
+        internal Dictionary<string, string> QueryParameters { get; private set; } = new Dictionary<string, string>();
+        internal Dictionary<string, string> Headers { get; private set; } = new Dictionary<string, string>();
+        internal HttpMethod Method { get; private set; }
+        internal string Resource { get; private set; }
+        internal string JsonBody { get; private set; }
+        internal System.IO.Stream FileStream { get; private set; }
+        internal string FileStreamContentType { get; private set; }
+        internal string FileStreamName { get; private set; }
 
-        public void AddHeader(string name, string value) => Headers[name] = value;
+        internal void AddHeader(string name, string value) => Headers[name] = value;
 
-        public void AddQueryParameter(string name, string value) => QueryParameters[name] = value;
+        internal void AddQueryParameter(string name, string value) => QueryParameters[name] = value;
 
-        public void SetJsonBody(string json) => JsonBody = json;
+        internal void SetJsonBody(string json) => JsonBody = json;
 
-        public void SetFileStream(System.IO.Stream stream, string name, string contentType)
+        internal void SetFileStream(System.IO.Stream stream, string name, string contentType)

```

Comment:
same here

Reply:
**the class was already internal** so no one could use it before. actually, I don't even understand this:

```csharp
internal class Myclass
{
  public void MyMethod()
  {
  }
}
```

Reply:
oops then doesn't make sense, actually compiler should spit some warnings for it